src/displaycontroller.h

```diff
@@ -539,6 +539,30 @@ struct Cursor {
 struct QuadTreeObject;
 
 
+enum PaintMode : uint8_t {
+  Set = 0,      // Plot colour
+  OR = 1,       // OR colour onto screen
+  AND = 2,      // AND colour onto screen
+  XOR = 3,      // XOR colour onto screen
+  Invert = 4,   // Invert colour on screen
+  NOT = 4,      // NOT colour onto screen, alias for Invert
+  NoOp = 5,     // No operation
+  ANDNOT = 6,   // AND colour on screen with NOT colour
+  ORNOT = 7,    // OR colour on screen with NOT colour
+};
+
+/**
+ * @brief Specifies general paint options.
+ */
+struct PaintOptions {
+  uint8_t swapFGBG : 1;  /**< If enabled swaps foreground and background colors */
+  uint8_t NOT      : 1;  /**< If enabled performs NOT logical operator on destination. Implemented only for straight lines and non-filled rectangles. */
+  PaintMode mode   : 3;  /**< Paint mode */
+
+  PaintOptions() : swapFGBG(false), NOT(false), mode(PaintMode::Set) { }
+} __attribute__ ((packed));
```

Comment:
these are the new paint modes

NOT is still there inside PaintOptions for compatibility only.  it's not used any more by the Console8 agon-vdp codebase

Reply:
How about changing the names to SET, INVERT, NO_OP, AND_NOT and OR_NOT? Make it more consistent.

Reply:
most of the enums in this codebase seem to use PascalCase for their naming.  I tried to follow suit here, however for the various painting modes that are named after boolean it felt more appropriate to name those in all caps, as that feels more conventional.

I hadn't considered putting underscores into the composite ANDNOT and ORNOT names.  there's a few uses of underscores in enum naming used in this codebase, so that could indeed be consistent and may be clearer